editor/src/core/property-controls/property-controls-utils.ts

```diff
@@ -354,65 +361,77 @@ export function getPropertyControlsForTargetFromEditor(
-  const jsxName = MetadataUtils.getJSXElementName(target, rootComponents, jsxMetadata)
-  if (importedName != null && tagName != null) {
-    // TODO default and star imports
-    let filename = Object.keys(openImports).find((key) => {
-      return pluck(openImports[key].importedFromWithin, 'name').includes(importedName)
-    })
-    if (filename == null && jsxName != null && isIntrinsicHTMLElement(jsxName)) {
-      /**
-       * We detected an intrinsic HTML Element (such as div, a, span, etc...)
-       * for the sake of simplicity, we assume here that they all support the style prop. if we need more detailed
-       * information for them, feel free to turn this into a real data structure that contains specific props for specific elements,
-       * but for now, I just return a one-size-fits-all PropertyControls result here
-       */
-      return HtmlElementStyleObjectProps
-    }
-    if (filename == null && openFilePath != null) {
-      filename = openFilePath.replace(/\.(js|jsx|ts|tsx)$/, '')
-    }
-    if (filename != null) {
-      // TODO figure out absolute filepath
-      const absoluteFilePath = filename.startsWith('.') ? `${filename.slice(1)}` : `${filename}`
-      if (
-        propertyControlsInfo[absoluteFilePath] != null &&
-        propertyControlsInfo[absoluteFilePath][tagName] != null
-      ) {
-        return propertyControlsInfo[absoluteFilePath][tagName] as PropertyControls
+  return withUnderlyingTarget(
+    target,
+    projectContents,
+    nodeModules,
+    openFilePath,
+    null,
+    (
+      success: ParseSuccess,
+      element: JSXElement,
+      underlyingTarget: StaticInstancePath,
+      underlyingFilePath: string,
+    ) => {
+      const importedFrom = importedFromWhere(
+        underlyingFilePath,
+        element.name.baseVariable,
+        success.topLevelElements,
+        success.imports,
+      )
+
+      let filenameForLookup: string | null = null
+      if (importedFrom == null) {
+        if (isIntrinsicHTMLElement(element.name)) {
+          /**
+           * We detected an intrinsic HTML Element (such as div, a, span, etc...)
+           * for the sake of simplicity, we assume here that they all support the style prop. if we need more detailed
+           * information for them, feel free to turn this into a real data structure that contains specific props for specific elements,
+           * but for now, I just return a one-size-fits-all PropertyControls result here
+           */
+          return HtmlElementStyleObjectProps
+        } else if (openFilePath != null) {
+          filenameForLookup = openFilePath.replace(/\.(js|jsx|ts|tsx)$/, '')
+        }
       } else {
+        filenameForLookup = importedFrom
+      }
+      if (filenameForLookup == null) {
         return null
+      } else {
+        // TODO figure out absolute filepath
+        const absoluteFilePath = filenameForLookup.startsWith('.')
```

Comment:
is this `TODO figure out absolute filepath` still relevant here? if not, I'd remove this `.` removal gibberish altogether.
But if it _is_ relevant, does that have multi-file implications for us?

Reply:
I'd just moved it around as I'd refactored the rest, but given the logic I'd somewhat assume it would be called `relativeFilePath`, as that's what it would imply. But I think now this should always be an absolute path anyway. Will look into it.

Reply:
Fixed in 3455633c.